mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,322 @@
+                GEOR.util.errorDialog({
+                    msg: tr("querier.layer.error")
+                });
+            },
+            scope: this
+        });
+
+        this.layer = new OpenLayers.Layer.Vector("__georchestra_filterbuilder", {
+            displayInLayerSwitcher: false,
+            styleMap: GEOR.util.getStyleMap({
+                "default": {
+                    strokeWidth: 2,
+                    strokeColor: "#ee5400",
+                    fillOpacity: 0
+                }
+            })
+        });
+
+        this.filterbuilder = new Styler.FilterBuilder({
+            defaultBuilderType: Styler.FilterBuilder.ALL_OF,
+            filterPanelOptions: {
+                attributesComboConfig: {
+                    displayField: "name",
+                    listWidth: 165,
+                    tpl: GEOR.util.getAttributesComboTpl()
+                }
+            },
+            attributes: this.attributeStore,
+            toolbarType: "tbar",
+            map: this.map,
+            allowGroups: true,
+            noConditionOnInit: false,
+            deactivable: true,
+            autoScroll: true,
+            allowSpatial: true,
+            cookieProvider: this.cookieProvider,
+            vectorLayer: this.layer
+        });
+        this.items = [this.filterbuilder];
+
+        this.buttons = [{
+            text: OpenLayers.i18n("Close"),
+            handler: this.close,
+            scope: this
+        }, {
+            text: OpenLayers.i18n("Search"),
+            handler: this.search,
+            scope: this
+        }];
+
+        this.addEvents(
+            /**
+             * @event searchresults
+             * Fires when we've received a response from server 
+             *
+             * Listener arguments:
+             * options - {Object} A hash containing response, model and format
+             */
+            "searchresults",
+
+            /**
+             * @event search
+             * Fires when the user presses the search button
+             *
+             * Listener arguments:
+             * panelCfg - {Object} Config object for a panel 
+             */
+            "search"
+        );
+
+        GEOR.Querier.superclass.initComponent.call(this);
+    },
+
+
+    /**
+     * Method: _checkFilter
+     * Checks that a filter is not missing items.
+     *
+     * Parameters:
+     * filter - {OpenLayers.Filter} the filter
+     *
+     * Returns:
+     * {Boolean} Filter is correct ?
+     */
+    _checkFilter: function(filter) {
+        var filters = filter.filters || [filter];
+        for (var i=0, l=filters.length; i<l; i++) {
+            var f = filters[i];
+            if (f.CLASS_NAME == "OpenLayers.Filter.Logical") {
+                if (!this._checkFilter(f)) {
+                    return false;
+                }
+            } else if (!(f.value && f.type && 
+                (f.property || f.CLASS_NAME == "OpenLayers.Filter.Spatial"))) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/georchestra/georchestra/file/2337895343/issues/source?bid=3200181&fileBranchId=3209168#l237)